cluster/store/meta_class.go

```diff
@@ -121,19 +122,38 @@ func (m *metaClass) CopyShardingState() *sharding.State {
 	return &st
 }
 
-func (m *metaClass) AddProperty(p models.Property) error {
+func (m *metaClass) AddProperty(props ...*models.Property) error {
 	m.Lock()
 	defer m.Unlock()
 
 	// update all at once to prevent race condition with concurrent readers
-	src := m.Class.Properties
-	dest := make([]*models.Property, len(src)+1)
-	copy(dest, src)
-	dest[len(src)] = &p
+	src := filterOutDup(m.Class.Properties, props)
+	dest := make([]*models.Property, len(src)+len(props))
+	copy(dest, append(src, props...))
 	m.Class.Properties = dest
 	return nil
 }
 
+// filterOutDup removes from the old any existing property could cause duplication
+func filterOutDup(old, new []*models.Property) []*models.Property {
+	// create memory to avoid duplication
+	var newUnique []*models.Property
+	mem := make(map[string]int, len(new))
```

Comment:
to be even more memory efficient at this task of finding duplicates you can use `map[string]struct{}` instead of `map[string]int`

Reply:
it's not random type, we need type int for the index of the item